File: data_structure/dy102/1874.java
import java.io.IOException;
import java.util.Scanner;
import java.util.Stack;

public class Main {
    public void solution(int[] arr) {
        Stack<Integer> stack = new Stack<>();
        StringBuilder sb = new StringBuilder();
        int count = 1;
        int index = 0;
        while (true) {
            if (index > arr.length - 1) { //배열의 길이를 넘어서면, 완료된 것
                break;
            }
            if (count > arr.length + 1) { //주어진 수열의 수를 넘어서면, 잘못된 것
                System.out.println("NO");
                return;
            }
            if (stack.isEmpty() || stack.peek() != arr[index]) { //스택이 비거나(예외방지), 꺼내서 쓸 수 없는 상태일 때
                stack.push(count);
                count++;
                sb.append("+").append("\n");
            } else {
                stack.pop();
                index++;
                sb.append("-").append("\n");
            }
        }
        System.out.println(sb);
    }

    public static void main(String[] args) throws IOException {
        Main main = new Main();
        Scanner scanner = new Scanner(System.in);

        int x = Integer.parseInt(scanner.nextLine());
        int[] arr = new int[x];
        for (int i = 0; i < x; i++) {
            arr[i] = scanner.nextInt();
        }
        main.solution(arr);
    }
}
//특정 수가 꼭대기에 존재할 때까지 push. 이후에 pop

//어떤 원리인지는 파악하기 쉬웠으나, 예외를 처리하는 부분이 까다로웠음.

File: data_structure/yuseok0215/1874.py
"""
여러번 봐줘야 하는 문제
count를 누적으로 이용해서 오름차순 순열을 표현했다
그리고 현재 숫자를 기준으로 푸쉬할만큼 하고 마지막 pop할 숫자가 현재 숫자와 일치하지 않으면
아예 못만든다는 사실을 인지하는 것이 필요했다.

43687521

43 6 87 521
9

1234 ++++
12 -- 
1256 ++
125 -
12578 ++
125 --
---

"""

count = 1
temp = True
stack = []
op = []

N = int(input())
for i in range(N):
    num = int(input())
    # num이하 숫자까지 스택에 넣기
    while count <= num: # 제일 중요한 부분....
        stack.append(count)
        op.append('+')
        count += 1

    # num이랑 스택 맨 위 숫자가 동일하다면 제거
    if stack[-1] == num:
        stack.pop()
        op.append('-')
    # 스택 수열을 만들 수 없으므로 NO
    else:
        temp = False
        break

# 스택 수열을 만들수 있는지 여부에 따라 출력 
if temp == False:
    print("NO")
else:
    for i in op:
        print(i)